armi/physics/neutronics/tests/test_crossSectionManager.py

```diff
@@ -849,6 +868,36 @@ def test_createRepresentativeBlocksUsingExistingBlocks(self):
         )
         self.assertEqual(origXSIDsFromNew["BA"], "AA")
 
+        # check that settings were copied correctly
+        baSettings = self.csm.cs[CONF_CROSS_SECTION]["BA"]
+        self.assertEqual(baSettings.xsID, "BA")
+        self.assertEqual(baSettings.geometry, aaSettings.geometry)
+        self.assertEqual(baSettings.xsFileLocation, aaSettings.xsFileLocation)
+        self.assertEqual(baSettings.validBlockTypes, aaSettings.validBlockTypes)
+        self.assertEqual(baSettings.blockRepresentation, aaSettings.blockRepresentation)
+        self.assertEqual(baSettings.fluxFileLocation, aaSettings.fluxFileLocation)
+        self.assertEqual(baSettings.driverID, aaSettings.driverID)
+        self.assertEqual(baSettings.criticalBuckling, aaSettings.criticalBuckling)
+        self.assertEqual(
+            baSettings.nuclideReactionDriver, aaSettings.nuclideReactionDriver
+        )
+        self.assertEqual(baSettings.externalDriver, aaSettings.externalDriver)
+        self.assertEqual(
+            baSettings.useHomogenizedBlockComposition,
+            aaSettings.useHomogenizedBlockComposition,
+        )
+        self.assertEqual(baSettings.numInternalRings, aaSettings.numInternalRings)
+        self.assertEqual(baSettings.numExternalRings, aaSettings.numExternalRings)
+        self.assertEqual(baSettings.mergeIntoClad, aaSettings.mergeIntoClad)
+        self.assertEqual(
+            baSettings.meshSubdivisionsPerCm, aaSettings.meshSubdivisionsPerCm
+        )
+        self.assertEqual(baSettings.xsMaxAtomNumber, aaSettings.xsMaxAtomNumber)
+        self.assertEqual(baSettings.minDriverDensity, aaSettings.minDriverDensity)
+        self.assertEqual(baSettings.averageByComponent, aaSettings.averageByComponent)
+        self.assertEqual(baSettings.xsExecuteExclusive, aaSettings.xsExecuteExclusive)
+        self.assertEqual(baSettings.xsPriority, aaSettings.xsPriority)
```

Comment:
Might be cleaner and easier to extend if this were done programmatically in a loop. Then it could be made so that it extends automatically when new sub-settings are added to `crossSectionControl`. That might only be practical if [`XSModelingOptions`](https://github.com/terrapower/armi/blob/f9903fe5a5d30ceda389d6494020f7a0c0f01b92/armi/physics/neutronics/crossSectionSettings.py#L328) were converted to a [`dataclass`](https://docs.python.org/3/library/dataclasses.html#dataclasses.dataclass) though.

Reply:
Not important!

Reply:
Good idea, I was able to use `xsSettings.__dict__` to simplify it significantly